tests/lib/collectors/gcloud_e2e.py

```diff
@@ -0,0 +1,253 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""End to end test for the Google Cloud Collector."""
+import json
+import os
+import time
+import unittest
+import logging
+
+from googleapiclient.errors import HttpError
+
+from libcloudforensics import gcp
+from dftimewolf import config
+from dftimewolf.lib import state
+from dftimewolf.lib.collectors import gcloud
+
+log = logging.getLogger()
+
+
+class EndToEndTest(unittest.TestCase):
+  """End to end test on GCP for the gcloud.py collector.
+
+  This end-to-end test runs directly on GCP and tests that:
+    1. The gcloud.py collector connects to the target instance and makes a
+    snapshot of the boot disk (by default) or of the disk passed in
+    parameter to the collector's SetUp method (disk_names).
+    2. A new disk is created from the taken snapshot.
+    3. If an analysis VM already exists, the module will attach the disk
+    copy to the VM. Otherwise, it will create a new GCP instance for
+    analysis purpose and attach the disk copy to it.
+
+  To run this test, add your project information to a project_info.json file:
+
+  {
+    "project_id": "xxx", # required
+    "instance": "xxx", # required
+    "disk": "xxx", # optional
+    "zone": "xxx" # required
+  }
+
+  Export a PROJECT_INFO environment variable with the absolute path to your
+  file: "user@terminal:~$ export PROJECT_INFO='absolute/path/project_info.json'"
+
+  """
+
+  def __init__(self, *args, **kwargs):
+    super(EndToEndTest, self).__init__(*args, **kwargs)
+    project_info = os.environ.get('PROJECT_INFO')
+
+    if project_info is None:
+      self.error_msg = 'Error: please make sure that you defined the ' \
+                       '"PROJECT_INFO" environment variable pointing ' \
+                       'to your project settings.'
+      return
+    try:
+      json_file = open(project_info)
+      project_info = json.load(json_file)
+      json_file.close()
+    except ValueError as exception:
+      self.error_msg = 'Error: cannot parse JSON file. {0:s}'.format(
+          str(exception))
+      return
+
+    if not all(key in project_info for key in ['project_id', 'instance',
+                                               'zone']):
+      self.error_msg = 'Error: please make sure that your JSON file ' \
+                       'has the required entries. The file should ' \
+                       'contain at least the following: ["project_id", ' \
+                       '"instance", "zone"].'
+      return
+
+    self.project_id = project_info['project_id']
+    self.instance_to_analyse = project_info['instance']
+    # Optional: test a disk other than the boot disk
+    if 'disk' in project_info:
+      self.disk_to_forensic = project_info['disk']
+    else:
+      self.disk_to_forensic = None
+    self.zone = project_info['zone']
+
+  def setUp(self):
+    if hasattr(self, 'error_msg'):
+      raise unittest.SkipTest(self.error_msg)
```

Comment:
Will `SkipTest` actually generate an error in the test run? Or will it say something like "yay tests succeeded, 3 / 3 tests skipepd"